bundles/tools.vitruv.testutils/src/tools/vitruv/testutils/views/ChangePublishingTestView.xtend

```diff
@@ -1,71 +1,105 @@
 import static extension edu.kit.ipd.sdq.commons.util.org.eclipse.emf.ecore.resource.ResourceSetUtil.withGlobalFactories
-import tools.vitruv.change.composite.propagation.ChangeableModelRepository
-import tools.vitruv.change.propagation.ChangePropagationSpecification
-import tools.vitruv.testutils.TestUserInteraction
-import tools.vitruv.change.propagation.ChangePropagationSpecificationRepository
-import static tools.vitruv.testutils.TestModelRepositoryFactory.createTestChangeableModelRepository
-import tools.vitruv.change.propagation.impl.DefaultChangeableModelRepository
 
 /**
  * A test view that will record and publish the changes created in it.
  */
 class ChangePublishingTestView implements NonTransactionalTestView {
 	val ResourceSet resourceSet
+	val UuidResolver uuidResolver
 	@Delegate
 	val TestView delegate
 	val ChangeRecorder changeRecorder
-	val List<(VitruviusChange)=>List<PropagatedChange>> changeProcessors = new LinkedList()
+	val ChangeableModelRepository modelRepository
+	val BiConsumer<Resource, UuidResolver> uuidResolution
 	var disposeViewResourcesAfterPropagation = true
 
 	/**
 	 * Creates a test view that will store its persisted resources in the
 	 * provided {@code persistenceDirectory}, allow to program interactions through the provided {@code userInteraction},
-	 * use the provided {@code uriMode}.
+	 * use the provided {@code uriMode} and be connected to the provided {@code virtualModel}.
+	 * 
+	 * @param persistenceDirectory is the directory to store files at.
+	 * @param userInteraction the {@link TestUserInteraction} to use for interactions during change propagation.
+	 * @param uriMode is the URI mode.
+	 * @param changeableModelRepository is the repository responsible for propagating and storing the models.
+	 * @param uuidResolution is a consumer that populates the given view's {@link UuidResolver} with the UUIDs of all elements in the given {@link Resource}.
 	 */
 	new(
 		Path persistenceDirectory,
-		tools.vitruv.testutils.TestUserInteraction userInteraction,
-		UriMode uriMode
+		TestUserInteraction userInteraction,
+		UriMode uriMode,
+		ChangeableModelRepository changeableModelRepository,
+		BiConsumer<Resource, UuidResolver> uuidResolution
 	) {
 		this.resourceSet = new ResourceSetImpl().withGlobalFactories()
+		this.uuidResolver = UuidResolver.create(resourceSet)
+		this.modelRepository = changeableModelRepository
 		this.delegate = new BasicTestView(persistenceDirectory, resourceSet, userInteraction, uriMode)
 		this.changeRecorder = new ChangeRecorder(resourceSet)
+		this.uuidResolution = uuidResolution
 		changeRecorder.beginRecording()
 	}
 
 	/**
 	 * Creates a test view that will store its persisted resources in the
 	 * provided {@code persistenceDirectory}, allow to program interactions through the provided {@code userInteraction},
 	 * use the provided {@code uriMode} and be connected to the provided {@code virtualModel}.
```

Comment:
Reference to`virtualModel` seams unclear to me.

Reply:
Agree, done in [0407629](https://github.com/vitruv-tools/Vitruv-Change/pull/47/commits/0407629023972f24a6133cde0a5c60f741193f1c)